corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/views/query.js

```diff
@@ -119,14 +121,26 @@ hqDefine("cloudcare/js/formplayer/menus/views/query", function () {
                 }
                 else {
                     // Set lookup table option by label
-                    var option = domElement.find("option").filter(function (_) {
-                        return $(this).text().trim() === value;
-                    });
-                    if (option.length === 1) {
-                        domElement.val(String(option.index() - 1)).trigger("change");
-                    } else {
-                        domElement.val("-1").trigger('change');
+                    var matchingOption = function (el) {
+                        return el.find("option").filter(function (_) {
+                            return $(this).text().trim() === value;
+                        });
                     }
+                    domElement.each(function () {
+                        var option = matchingOption($(this));
+                        if (this.multiple === true) {
+                            if (option.length === 1 && $(this).val().indexOf(option.val()) === -1) {
+                                $(this).val($(this).val().concat(option.val())).trigger("change");
```

Comment:
Is this string concat or array concat? 

Shouldn't this be adding the option index (converted to zero index) instead of the value? Or is the option value already the index (looks like it is based on the HTML but the select1 version uses `option.index() - 1`

Reply:
Also, should the trigger be called for each change or just once?

Reply:
>  Or is the option value already the index

Yes, that's right. The [template](https://github.com/dimagi/commcare-hq/pull/29808/files#diff-5b5306df02c3e60efa4688371a6f46de4e1f3733d92b9bf0b3e6f17917ace93cR50) uses the indices as values.

> Also, should the trigger be called for each change or just once?

It will trigger for each field once.